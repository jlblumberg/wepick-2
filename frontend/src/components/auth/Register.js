import React, { useState } from 'react';
import { BrowserRouter, Link } from 'react-router-dom';

function Register() {

  const [formValues, setFormValues] = useState({
    name: "",
    email: "",
    password: "",
    password2: "",
    errors: { name: "", email: "", password: "", password2: "" }
  })

  const onChange = (e) => {
    setFormValues({ ...formValues, [e.target.id]: e.target.value })
  }

  return (
    <div>
      <BrowserRouter>
        <p id="alreadyregistered">Already have an account? <Link id="login" to="/login">Log In</Link></p>
      </BrowserRouter>
      <form>
        <div>
          <label htmlFor="name">Name</label>
          <input
            id='name'
            value={formValues.name}
            onChange={onChange}
            type="text"
            errors={formValues.errors.name}
          />
          <label htmlFor="email">Email</label>
          <input
            id='email'
            value={formValues.email}
            onChange={onChange}
            type="email"
            errors={formValues.errors.email}
          />
          <label htmlFor="password">Password</label>
          <input
            id='password'
            value={formValues.password}
            onChange={onChange}
            type="password"
            errors={formValues.errors.password}
          />
          <label htmlFor="password2">Confirm Password</label>
          <input
            id='password2'
            value={formValues.password2}
            onChange={onChange}
            type="password"
            errors={formValues.errors.password2}
          />
        </div>
        <button id="sign-up"
          style={{
            width: "150px",
            borderRadius: "3px",
            letterSpacing: "1.5px",
            marginTop: "1rem"
          }}
          type="submit"
          className="btn btn-large waves-effect waves-light hoverable blue accent-3"
        >Sign up</button>
      </form>
    </div>
  )
}

export default Register;
