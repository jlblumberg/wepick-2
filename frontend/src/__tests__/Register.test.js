import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import Register from '../components/auth/Register';

Enzyme.configure({ adapter: new Adapter() });

describe('Register', () => {
  const wrapper = shallow(<Register/>);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders an already have account log in link', () => {
    expect(wrapper.find('#alreadyregistered').text()).toContain('Already have an account?');
    expect(wrapper.find('#login').text()).toEqual('Log In');
  });

  it('has a field for name', () => {
    expect(wrapper.find('input#name').prop('value')).toEqual("");
    expect(wrapper.find('input#name').prop('type')).toEqual("text");
    expect(wrapper.find('input#name').prop('errors')).toEqual("");
  });
  
  it('has a field for email', () => {
    expect(wrapper.find('input#email').prop('value')).toEqual("");
    expect(wrapper.find('input#email').prop('type')).toEqual("email");
    expect(wrapper.find('input#email').prop('errors')).toEqual("");
  });

  it('has a field for password', () => {
    expect(wrapper.find('input#password').prop('value')).toEqual("");
    expect(wrapper.find('input#password').prop('type')).toEqual("password");
    expect(wrapper.find('input#password').prop('errors')).toEqual("");
  });

  it('has a field for password2', () => {
    expect(wrapper.find('input#password2').prop('value')).toEqual("");
    expect(wrapper.find('input#password2').prop('type')).toEqual("password");
    expect(wrapper.find('input#password2').prop('errors')).toEqual("");
  });

  it('fields can be updated', () => {
    wrapper.find({ id: 'name' }).simulate('change', { target: { id: 'name', value: 'Mike'} });
    expect(wrapper.find({ id: 'name' }).prop('value')).toEqual('Mike');
    wrapper.find({ id: 'email' }).simulate('change', { target: { id: 'email', value: 'mike@example.com'} });
    expect(wrapper.find({ id: 'email' }).prop('value')).toEqual('mike@example.com');
    wrapper.find({ id: 'password' }).simulate('change', { target: { id: 'password', value: 'examplepassword'} });
    expect(wrapper.find({ id: 'password' }).prop('value')).toEqual('examplepassword');
    wrapper.find({ id: 'password2' }).simulate('change', { target: { id: 'password2', value: 'examplepassword'} });
    expect(wrapper.find({ id: 'password2' }).prop('value')).toEqual('examplepassword');
  });

  it('has a submit button', () => {
    expect(wrapper.find('button#sign-up').text()).toEqual('Sign up');
  });

});