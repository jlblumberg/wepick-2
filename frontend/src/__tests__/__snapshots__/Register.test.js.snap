// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register renders correctly 1`] = `ShallowWrapper {}`;
